### PR TITLE
Fix image name validation

### PR DESCRIPTION
**Pull request by qbarrand**
Fix a bug that would validate a registry hostname with port as a qualified image name with tag.
Make registry hostname and identifier (tag or digest) mandatory in `containerImage` fields of the Module CR.

/cc @mresvanis @ybettan @yevgeny-shnaidman

### CODE REVIEW

File: api/v1beta1/module_webhook_test.go [hunk 1]
```diff
@@ -40,6 +40,8 @@ func getLengthAfterSlash(s string) int {
 	return len(before)
 }
 
+const validImageName = "example.com/image:tag"
```
Comment (ybettan) [inline api/v1beta1/module_webhook_test.go:43]:
```suggestion
const validImageName = "example.com/user/image:tag"
```

Comment (ybettan) [reply api/v1beta1/module_webhook_test.go:43]: Not crucial if you don't like it.

**Comment by qbarrand** [comment]
Will postpone this change to v2.1.
/hold

**Pull request closed without merging**